### DISCUSSION TITLE
Writing a JSON-RPC method in version 5

### DISCUSSION THREAD

**Discussion by bcb** [bcb/jsonrpcserver#147] [General]
Version 5 is in the `master` branch.

Development of version 4 has moved to the `4.x` branch.

**Comment by bcb** [comment]
Currently planning for the following usage for writing methods:
```python
from jsonrpcserver import method
from jsonrpcserver.dispatcher import Context
from jsonrpcserver.response import InvalidParamsResponse, Response, SuccessResponse

@method
def fruits(context: Context, color: str) -> Response:
    if color not in ["red", "orange", "yellow"]:
        return InvalidParamsResponse("No fruits of that colour", id=context.request.id)
    return SuccessResponse("blue", id=context.request.id)
```

This give the method writer complete control over the response going back to the client.

Though I'm not sure about the "id" part, it's annoying to have to specify that every time.

**Reply by st31ny** [reply]
Looks good to me, except that setting the ID should be optional. Some level of abstraction is the key point of having a library in the first place.

**Reply by bcb** [reply]
@st31ny yeah I agree, I'll see if the library can handle the id part.

**Reply by bcb** [reply]
Let's move this discussion to #158
